auslib/test/admin/views/test_required_signoffs.py

```diff
@@ -0,0 +1,923 @@
+        self.assertEquals(db_data, expected)
+
+        base_row = dbo.productRequiredSignoffs.t.select().where(dbo.productRequiredSignoffs.channel == "j")\
+                                                         .where(dbo.productRequiredSignoffs.product == "fake")\
+                                                         .where(dbo.productRequiredSignoffs.role == "releng")\
+                                                         .execute().fetchall()
+        self.assertEquals(len(base_row), 0)
+
+    def testGetScheduledChangeHistoryRevisions(self):
+        ret = self._get("/scheduled_changes/required_signoffs/product/3/revisions")
+        self.assertEquals(ret.status_code, 200, ret.data)
+        expected = {
+            "count": 2,
+            "revisions": [
+                {
+                    "change_id": 7, "changed_by": "bill", "timestamp": 100, "sc_id": 3, "scheduled_by": "bill", "change_type": "insert",
+                    "data_version": None, "product": "fake", "channel": "e", "role": "releng", "signoffs_required": 1,
+                    "when": 300000000, "complete": True, "sc_data_version": 2,
+                },
+                {
+                    "change_id": 6, "changed_by": "bill", "timestamp": 81, "sc_id": 3, "scheduled_by": "bill", "change_type": "insert",
+                    "data_version": None, "product": "fake", "channel": "e", "role": "releng", "signoffs_required": 2,
+                    "when": 300000000, "complete": False, "sc_data_version": 1,
+                },
+            ],
+        }
+        self.assertEquals(json.loads(ret.data), expected)
+
+    @mock.patch("time.time", mock.MagicMock(return_value=100))
+    def testSignoffWithPermission(self):
+        ret = self._post("/scheduled_changes/required_signoffs/product/2/signoffs", data=dict(role="relman"), username="bob")
+        self.assertEquals(ret.status_code, 200, ret.data)
+        r = dbo.productRequiredSignoffs.scheduled_changes.signoffs.t.select()\
+            .where(dbo.productRequiredSignoffs.scheduled_changes.signoffs.sc_id == 2).execute().fetchall()
+        self.assertEquals(len(r), 2)
+        self.assertEquals(dict(r[0]), {"sc_id": 2, "username": "bill", "role": "releng"})
+        self.assertEquals(dict(r[1]), {"sc_id": 2, "username": "bob", "role": "relman"})
+        r = dbo.productRequiredSignoffs.scheduled_changes.signoffs.history.t.select()\
+            .where(dbo.productRequiredSignoffs.scheduled_changes.signoffs.history.sc_id == 2).execute().fetchall()
+        self.assertEquals(len(r), 4)
+        self.assertEquals(dict(r[0]), {"change_id": 3, "changed_by": "bill", "timestamp": 60, "sc_id": 2, "username": "bill", "role": None})
+        self.assertEquals(dict(r[1]), {"change_id": 4, "changed_by": "bill", "timestamp": 61, "sc_id": 2, "username": "bill", "role": "releng"})
+        self.assertEquals(dict(r[2]), {"change_id": 10, "changed_by": "bob", "timestamp": 99999, "sc_id": 2, "username": "bob", "role": None})
```

Comment:
Is it the behavior that we talked about yesterday? The one where we're force to stack insertions in the database?